Library/Formula/rust.rb

```diff
@@ -23,6 +30,26 @@ def install
     system "./configure", *args
     system "make"
     system "make install"
+
+    resource("cargo").stage do
+      # as soon as cargo can be built without gated features,
+      # running ./travis.install.deps.sh should be useless
+      # and --local-rust-root should be able to point to `prefix`
+
+      args = ["--prefix=#{prefix}"]
+      if build.head?
+        args << "--local-rust-root=\"#{prefix}\""
+      else
+        system "./.travis.install.deps.sh"
+        args << "--local-rust-root=\"$PWD\"/rustc"
+      end
+      system "./configure", *args
+      system "make"
```

Comment:
This step runs [`src/etc/dl-snapshot.py`](https://github.com/rust-lang/cargo/blob/master/src/etc/dl-snapshot.py), which downloads a cargo snapshot since cargo needs itself to build. It reads the topmost entry in `src/snapshots.txt` to find the date and SHA1 hash (per platform) of the latest snapshot tarball. We can handle the situation in almost the same way as with `.travis.install.deps.sh` and manage this download as a resource. There is some renaming that takes place but nothing earth shattering: it's basically `s/cargo-nightly-whatever-platform/cargo/`.